glooshot/docs/tutorial/bookinfo_tutorial.md

```diff
@@ -0,0 +1,265 @@
+    timeStarted: "2019-05-13T17:27:34.650136785Z"
+```
+- Note that the state reports the experiment has "Failed". This is because the experiment was terminated because a threshold value was exceeded. If the experiment had been terminiated by a timeout, it would be in state "Succeeded".
+  - Experiments that fail, such as this one, indicate that our service is not as robust as we would like.
+- The experiment also reports the exact value that was observed, which caused the failure. Note that the value is 20, which exceeds our limit of 10. The reason for this is that Prometheus gathers metrics every 15 seconds. The metric value may rise above the limit in the time it takes for Prometheus to report the exceeded limit.
+
+### Repeat the experiment on a new version of the app
+- Now that we found a weakness in our app, let's fix it.
+- Let's deploy a version of the app that does not have this vulnerability. Instead of failing when no data is returned from the ratings service, the more robust version of our app will just exclude the ratings content.
+- In this demo, we happened to already have deployed this version of the app. Let's use Supergloo to update Istio so that all traffic is routed to the robust version of the app, as we did above.
+```bash
+kubectl delete routingrule -n glooshot reviews-v4
+supergloo apply routingrule trafficshifting \
+    --namespace glooshot \
+    --name reviews-v3 \
+    --dest-upstreams glooshot.default-reviews-9080 \
+    --target-mesh glooshot.istio-istio-system \
+    --destination glooshot.default-reviews-v3-9080:1
+```
+
+- Verify that the new routing rule was applied
+  - Refresh the page, you should see no errors
+  - Run the following command, you should see `reviews-v3` in the `glooshot` namespace
+```bash
+kubectl get routingrule --all-namespaces
+```
+
+- Now let's execute this experiment again to verify that our app is robust to failure.
+- This time, we do not expect any failures so we will set a shorter timeout.
+- We also need to increase the threshold, since we increased our metrics in the last experiment.
+- Use the following command to create a new experiment:
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: glooshot.solo.io/v1
+kind: Experiment
+metadata:
+  name: abort-ratings-metric-repeat
+  namespace: default
+spec:
+  spec:
+    duration: 30s
+    failureConditions:
+      - prometheusTrigger:
+          customQuery: |
+            scalar(sum(istio_requests_total{ source_app="productpage",response_code="500"}))
+          thresholdValue: 60
+          comparisonOperator: ">"
+    faults:
+    - destinationServices:
+      - name: default-ratings-9080
+        namespace: glooshot
+      fault:
+        abort:
+          httpStatus: 500
+        percentage: 100
+    targetMesh:
+      name: istio-istio-system
+      namespace: glooshot
+EOF
+```
+
+- Refresh the page, you should now see content from the reviews service and an error from the ratings service only.
```

Comment:
I didn't see that. Everything continued to work. I checked the experiment and it seemed to also fail out:

```
ceposta@postamaclab(~) $ k get exp abort-ratings-metric-repeat -o yaml
apiVersion: glooshot.solo.io/v1
kind: Experiment
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"glooshot.solo.io/v1","kind":"Experiment","metadata":{"annotations":{},"name":"abort-ratings-metric-repeat","namespace":"default"},"spec":{"spec":{"duration":"30s","failureConditions":[{"prometheusTrigger":{"comparisonOperator":"\u003e","customQuery":"scalar(sum(istio_requests_total{ source_app=\"productpage\",response_code=\"500\"}))\n","thresholdValue":60}}],"faults":[{"destinationServices":[{"name":"default-ratings-9080","namespace":"glooshot"}],"fault":{"abort":{"httpStatus":500},"percentage":100}}],"targetMesh":{"name":"istio-istio-system","namespace":"glooshot"}}}}
  creationTimestamp: "2019-05-16T18:51:57Z"
  generation: 3
  name: abort-ratings-metric-repeat
  namespace: default
  resourceVersion: "4913"
  selfLink: /apis/glooshot.solo.io/v1/namespaces/default/experiments/abort-ratings-metric-repeat
  uid: ae6b3acb-780b-11e9-8947-06b9bcf3d395
spec:
  result:
    failureReport:
      comparison_operator: '>'
      failure_type: value_exceeded_threshold
      threshold: "60"
      value: "120"
    state: Failed
    timeFinished: "2019-05-16T18:52:03.459504607Z"
    timeStarted: "2019-05-16T18:51:58.412798234Z"
  spec:
    duration: 30s
    failureConditions:
    - prometheusTrigger:
        comparisonOperator: '>'
        customQuery: |
          scalar(sum(istio_requests_total{ source_app="productpage",response_code="500"}))
        thresholdValue: 60
    faults:
    - destinationServices:
      - name: default-ratings-9080
        namespace: glooshot
      fault:
        abort:
          httpStatus: 500
        percentage: 100
    targetMesh:
      name: istio-istio-system
      namespace: glooshot
status: {}
```